syntax = "proto2";

package X;

message KeyRangeProto {
  optional uint64 low = 1;
  optional uint64 high = 2;
};

message NodeProto {
  optional uint32 nid = 1;
  optional string name = 2;
  optional string public_name = 3;
  optional uint32 port = 4;
  optional KeyRangeProto range = 5;
};

message ClusterProto {
  repeated NodeProto nodes = 1;
};